hpOneView/image_streamer/resources/artifact_bundles.py

```diff
@@ -0,0 +1,375 @@
+
+    def get_backups_by_id(self, id_or_uri):
+        """
+        Get the details of the backup for an Artifact Bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+
+        Returns:
+            list: Backup for an Artifacts Bundle.
+        """
+        uri = self.URI + self.BACKUPS_PATH + '/' + extract_id_from_uri(id_or_uri)
+        return self._client.get(id_or_uri=uri)
+
+    def download_archive_artifact_bundle(self, id_or_uri, file_path):
+        """
+        Download archive for the Artifact Bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+            file_path(str): Destination file path.
+
+        Returns:
+            list: A list of Artifacts Bundle.
+        """
+
+        uri = self.URI + self.BACKUPS_PATH + self.ARCHIVE_PATH + '/' + extract_id_from_uri(id_or_uri)
+
+        with open(file_path, 'wb') as file:
+            return self._connection.download_to_stream(file, uri)
+
+    def download_artifact_bundle(self, id_or_uri, file_path):
+        """
+        Download the Artifact Bundle.
+
+        Args:
+            id_or_uri: ID or URI of the Artifact Bundle.
+            file_path(str): Destination file path.
+
+        Returns:
+            list: A list of Artifacts Bundle.
+        """
+        uri = self.URI + self.DOWNLOAD_PATH + '/' + extract_id_from_uri(id_or_uri)
+
+        with open(file_path, 'wb') as file:
+            return self._connection.download_to_stream(file, uri)
+
+    def create_backup(self, deployment_groups_id_or_uri, timeout=-1):
+        """
+        Create Backup for Artifacts Bundle specified in the Deployment Groups.
+
+        Args:
+            deployment_groups_id_or_uri: ID or URI of the Deployment Groups.
+            timeout:
+                Timeout in seconds. Wait for task completion by default. The timeout does not abort the operation in
+                OneView, just stops waiting for its completion.
+
+        Returns:
+            dict: The Artifact Bundle.
+        """
+        if self.DEPLOYMENT_GROUPS_URI not in deployment_groups_id_or_uri:
+            deployment_groups_uri = self.DEPLOYMENT_GROUPS_URI + deployment_groups_id_or_uri
+        else:
+            deployment_groups_uri = deployment_groups_id_or_uri
+
+        data = {
+            "deploymentGroupURI": deployment_groups_uri
+        }
+
+        uri = self.URI + self.BACKUPS_PATH
+        return self._client.create(data, uri=uri, timeout=timeout)
+
+    def upload_bundle_from_file(self, file_path):
+        """
+        Restore an Artifact Bundle from backup file.
+
+        Args:
+            file_path (str): The File Path to restore the Artifact Bundle.
+
+        Returns:
+            dict: The Artifact Bundle.
+        """
+
+        upload_file_name = os.path.basename(file_path)
+
+        uri = self.URI
+
+        response, body = self._connection.post_multipart(uri, None, file_path, upload_file_name)
+        if response.status >= 400:
```

Comment:
I believe it could be useful for any resource, I suggest move this validation to the post_multipart method.